src/net.h

```diff
@@ -251,7 +252,12 @@ class CConnman
     void SetServices(const CService &addr, ServiceFlags nServices);
     void MarkAddressGood(const CAddress& addr);
     void AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddress& addrFrom, int64_t nTimePenalty = 0);
-    std::vector<CAddress> GetAddresses();
+
+    // Cache is used to minimize topology leaks, so it should
+    // be used for all non-trusted calls, for example, p2p.
+    // A non-malicious call (from RPC or a whitelisted node)
+    // should avoid using the cache by passing nullopt.
+    std::vector<CAddress> GetAddresses(Optional<Network> requestor_network);
```

Comment:
It feels to me like this would be more natural as an overloaded function rather than passing an optional:

- `GetAddresses()` returns an uncached vector of addresses from addrman as now
- `GetAddresses(Network)` returns the cached vector of addresses for that network.

Why? Because the logic for those two cases is entirely separated in the implementation, it avoids packing and unpacking the `Network` in an optional, and doing this means that some of the calling sites don't need to be changed.

Reply:
Agreed, will update.